playbooks/e2e.yaml

```diff
@@ -0,0 +1,73 @@
+---
+- name: E2E test
+  hosts: all
+  vars:
+    cloud_provider: "{{ cloud }}"  # inherited from the parent job
+    project_dir: "{{ ansible_user_dir }}/{{ zuul.project.src_dir }}"
+    project_tf_dir: "{{ project_dir }}/terraform"
+    prefix: "pr{{ zuul.change }}"
+    testcluster_name: "pr{{ zuul.change }}"
+  environment:
+    ENVIRONMENT: "{{ cloud_provider }}"
+  tasks:
+  - name: Prepare environment
+    block:
+    - name: Get PR details
+      uri:
+        url: "https://api.github.com/repos/{{ zuul.project.name }}/pulls/{{ zuul.change }}"
+        body_format: json
+        headers:
+          Accept: application/vnd.github+json
+          X-GitHub-Api-Version: 2022-11-28
+      register: pull_request
+    - name: Set facts based on PR details
+      set_fact:
+        git_branch_name: "{{ pull_request.json.head.ref }}"
+        git_repository_url: "{{ pull_request.json.head.repo.clone_url }}"
+    - name: Checkout PR feature branch {{ git_branch_name }}
```

Comment:
Will this checkout work, when the branch is from the fork?

Reply:
Good point! Yes, this won't work when PR is created from the fork, because the upstream repository does not contain a feature branch from the fork.   

This should be improved.

Reply:
see b3ec1973a526e8e3ad81263bbc79e2c8422980c0, 

Based on further investigation we (thx @chess-knight) realized that the Zuul pre-run role [prepare-workspace](https://github.com/SovereignCloudStack/zuul-config/blob/main/playbooks/base/pre.yaml#L10) rsync all PR changes to the working directory on all Zuul nodes. Hence, it is not necessary to checkout the feature branch.